activestorage/test/models/variant_test.rb

```diff
@@ -80,4 +97,20 @@ class ActiveStorage::VariantTest < ActiveSupport::TestCase
     variant = blob.variant(font: "a" * 10_000).processed
     assert_operator variant.service_url.length, :<, 525
   end
+
+  test "works for vips processor" do
+    begin
+      ActiveStorage.variant_processor = :vips
+      blob = create_file_blob(filename: "racecar.jpg")
+      variant = blob.variant(thumbnail_image: 100).processed
+
+      image = read_image(variant)
+      assert_equal 100, image.width
+      assert_equal 67, image.height
+    rescue LoadError
+      # libvips not installed
+    ensure
+      ActiveStorage.variant_processor = :mini_magick
+    end
+  end
```

Comment:
This test is failing in CI:

```
Error:
ActiveStorage::VariantTest#test_works_for_vips_processor:
NoMethodError: undefined method `source' for #<String:0x00000000043b5728>
```

Can you take a look?

Reply:
Fixed in https://github.com/rails/rails/pull/32471/commits/f2e2cef15bdb31353aee2254ca2ab378979cc24a. Requiring `ruby-vips` will raise a `LoadError` if libvips is not installed, and the `#processor` method was accidentally swallowing it.